### PR TITLE
Raise Kueue version in CodeFlare operator

### PR DESCRIPTION
**Pull request by sutaakar**
# Issue link

N/A

# What changes have been made

Raise Kueue version in CodeFlare operator to use latest Kueue dependency available in ODH.

# Verification steps

E2E tests pass.

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/openshift/client-go v0.0.0-20221019143426-16aed247da5c
 	github.com/project-codeflare/appwrapper v0.27.0
 	github.com/project-codeflare/codeflare-common v0.0.0-20240930133152-11fd6e3be6b3
-	github.com/ray-project/kuberay/ray-operator v1.1.1
+	github.com/ray-project/kuberay/ray-operator v1.2.1
```
Comment (dgrove-oss) [inline go.mod:15]: Why did this change?

File: go.mod [hunk 2]
```diff
@@ -35,11 +35,8 @@ replace go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp => go.open
 
 replace github.com/jackc/pgx/v4 => github.com/jackc/pgx/v5 v5.5.4
 
-// This replace directive supports the backlevel go version required by the ODH build
-replace github.com/project-codeflare/appwrapper v0.27.0 => github.com/project-codeflare/appwrapper v0.27.1-0.20241105160939-f71c6326acce
-
-// These replace directives deal with the backlevel ODH kueue version
-replace sigs.k8s.io/kueue v0.8.3 => github.com/opendatahub-io/kueue v0.7.0-odh-2
+// These replace directives deal with the backlevel KubeRay version compatible with CFO
+replace github.com/ray-project/kuberay/ray-operator v1.2.1 => github.com/ray-project/kuberay/ray-operator v1.1.1
```
Comment (dgrove-oss) [inline go.mod:39]: Why did this change?

**Review by dgrove-oss — COMMENTED** [review]
I'd suggest pulling the change in ray version out of this PR.  My guess is that is the source of the problem.

File: go.mod [hunk 3]
```diff
@@ -35,11 +35,8 @@ replace go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp => go.open
 
 replace github.com/jackc/pgx/v4 => github.com/jackc/pgx/v5 v5.5.4
 
-// This replace directive supports the backlevel go version required by the ODH build
-replace github.com/project-codeflare/appwrapper v0.27.0 => github.com/project-codeflare/appwrapper v0.27.1-0.20241105160939-f71c6326acce
-
-// These replace directives deal with the backlevel ODH kueue version
-replace sigs.k8s.io/kueue v0.8.3 => github.com/opendatahub-io/kueue v0.7.0-odh-2
```
Comment (dgrove-oss) [inline go.mod:42]: you probably still need a replace directive for kueue, just to a different odh version.

File: go.mod (hunk 1 above)
Comment (sutaakar) [reply go.mod:15]: This is brought by Kueue 0.8.3 dependency. That is why I had to rewrite it below.

File: go.mod (hunk 3 above)
Comment (sutaakar) [reply go.mod:42]: ok, will try

File: go.mod (hunk 2 above)
Comment (dgrove-oss) [reply go.mod:39]: ahh got it.  I read the diff backwards.

File: go.mod (hunk 3 above)
Comment (sutaakar) [reply go.mod:42]: Done

**Comment by sutaakar** [comment]
@dgrove-oss Please take a look.
Once merged we can upgrade AppWrapper.

**Review by astefanutti — APPROVED** [review]

**Pull request merged**